### PR TITLE
Add validation for fasta input and tests

### LINKED ISSUE #17: Add validation for fasta file input
### Description of feature

Currently the `schema_input.json` only validates for `query`, it would be nice to also validate for `fasta` as a path.

### PR DESCRIPTION
**Pull request by JoseEspinosa**
* Validate `fasta` input.
* Branching `samplesheet` into `samplesheet_query` and `samplesheet_fasta` and refactor code accordingly.
* Add test for `fasta`, run with `-profile test_fasta`.
* Run `test_fasta` on CI.
* Other minor fixes.

Closes #17 
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: bin/fetch_oma_by_sequence.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ def main() -> None:
         raise ValueError("Not enough arguments. Usage: fetch_oma_by_sequence.py <fasta> <id_out> <taxid_out> <exact_out>")
 
     seqs = SeqIO.parse(sys.argv[1], "fasta")
+
```
Comment (JoseEspinosa) [inline bin/fetch_oma_by_sequence.py]:
```suggestion
```

File: bin/fetch_oma_by_sequence.py [hunk 2]
```diff
@@ -53,6 +54,7 @@ def main() -> None:
                 raise ValueError("Isoform not found")
 
     print(entry["canonicalid"], file=open(sys.argv[2], "w"))
+
```
Comment (JoseEspinosa) [inline bin/fetch_oma_by_sequence.py]:
```suggestion
```

**Review by itrujnara — COMMENTED** [review]
The code looks good, hopefully it is tested as I have very little knowledge of nf-schema. Should we not update the readme and usage to reflect the changes in the input format?

File: README.md [hunk 3]
```diff
@@ -44,21 +44,20 @@ Steps that follow can be skipped with `--skip_downstream` in batch analysis.
...
 
-```csv
+```csv title="samplesheet.csv"
 id,query
 BicD2,Q8TD16
 ```
 
-If using the latter format, you must set `--uniprot_query` to true.
+> [!NOTE]
+> If you provide both a FASTA file and a UniProt ID only the later will be used.
```
Comment (itrujnara) [inline README.md]:
```suggestion
> If you provide both a FASTA file and a UniProt ID only the latter will be used.
```
Typo

File: docs/usage.md [hunk 4]
```diff
@@ -18,26 +18,27 @@ You will need to create a samplesheet with information about the samples you wou
...
-| `query` | The query of the user-specified type. If `--uniprot_query` is `true`, it should be a valid Uniprot accession. Otherwise, it should be a valid path to a FASTA file. |
+| Column  | Description                                                                                                                                       |
+| ------- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `id`    | User-defined identifier. It is used to identify output files for the protein. Can be anything descriptive, as long as it does not contain spaces. |
+| `query` | The query of the user-specified type. It should be a valid Uniprot accession.                                                                     |
+| `fasta` | It should be a valid path to a FASTA file.                                                                                                        |
 
 An [example samplesheet](../assets/samplesheet.csv) has been provided with the pipeline.
```
Comment (itrujnara) [inline docs/usage.md]: Do we want an example for FASTA samplesheet as well?

**Review by itrujnara — COMMENTED** [review]
Looks good, just some very minor comments

File: docs/usage.md (hunk 4 above)
Comment (JoseEspinosa) [reply docs/usage.md]: It is changed in line 33, or are you referring to something else?

Comment (itrujnara) [reply docs/usage.md]: Line 43 mentions a separate samplesheet file, I wonder if we need a similar one for FASTA. Just a question to consider, not necessarily demanding it.

Comment (JoseEspinosa) [reply docs/usage.md]: Done in [195ab8d](https://github.com/nf-core/reportho/pull/20/commits/195ab8d1f553887dac4a34d9a4f26e271f771322)

Comment (itrujnara) [inline docs/usage.md]:
```suggestion
An [example Uniprot samplesheet](../assets/samplesheet.csv) and [example FASTA samplesheet](../assets/samplesheet_fasta.csv) has been provided with the pipeline.
```

**Review by itrujnara — COMMENTED** [review]
Final tiny tweak

**Review by itrujnara — APPROVED** [review]
LGTM

**Pull request merged**